application-diagram-api/src/test/java/com/xwiki/diagram/internal/handlers/TestDiagramMacroSolrMetadataExtractor.java

```diff
@@ -0,0 +1,149 @@
+     * keep the reference so we don't break the macros that already exist.
+     */
+    @Test
+    public void updateReferenceCase1() throws XWikiException
+    {
+        MacroBlock block = this.createMacroBlock("Strategy");
+        when(this.xwikiDocument.getXDOM().getBlocks(any(), any())).thenReturn(List.of(block));
+        when(explicitDocumentReferenceResolver.resolve("Strategy", xwikiDocument.getDocumentReference())).thenReturn(
+            documentReference);
+        when(xwikiContext.getWiki().exists(documentReference, xwikiContext)).thenReturn(true);
+        when(entityNameValidationManager.getEntityReferenceNameStrategy().isValid("Strategy")).thenReturn(false);
+        when(entityNameValidationManager.getEntityReferenceNameStrategy().transform("Strategy")).thenReturn(
+            "Strategy2");
+        extractor.extract(this.xwikiDocument, this.solrDocument);
+        assertEquals("Strategy", block.getParameter(REFERENCE));
+    }
+
+    /**
+     * The macro has the default settings where the reference is missing and there is a naming strategy that updates
+     * the name.
+     *
+     */
+    @Test
+    public void updateReferenceCase3() throws XWikiException
+    {
+        MacroBlock block = this.createMacroBlock(null);
+        when(this.xwikiDocument.getXDOM().getBlocks(any(), any())).thenReturn(List.of(block));
+        when(explicitDocumentReferenceResolver.resolve("Strategy", xwikiDocument.getDocumentReference())).thenReturn(
+            documentReference);
+        when(xwikiContext.getWiki().exists(documentReference, xwikiContext)).thenReturn(false);
+        when(entityNameValidationManager.getEntityReferenceNameStrategy().isValid("Diagram")).thenReturn(false);
+        when(entityNameValidationManager.getEntityReferenceNameStrategy().transform("Diagram")).thenReturn("Anagram");
+        extractor.extract(this.xwikiDocument, this.solrDocument);
+        assertEquals("Anagram", block.getParameter(REFERENCE));
+    }
+
+    /**
+     * The diagram was created, it respects the name strategy, but the reference doesn't respect the name strategy.
+     */
+    @Test
+    public void updateReferenceCase2() throws XWikiException
+    {
+        MacroBlock block = this.createMacroBlock("AAAA");
+        when(this.xwikiDocument.getXDOM().getBlocks(any(), any())).thenReturn(List.of(block));
+        when(explicitDocumentReferenceResolver.resolve("AAAA", xwikiDocument.getDocumentReference())).thenReturn(
+            documentReference);
+        when(xwikiContext.getWiki().exists(documentReference, xwikiContext)).thenReturn(false);
+        when(entityNameValidationManager.getEntityReferenceNameStrategy().isValid("AAAA")).thenReturn(false);
+        when(entityNameValidationManager.getEntityReferenceNameStrategy().transform("AAAA")).thenReturn("BBBB");
+        extractor.extract(this.xwikiDocument, this.solrDocument);
+        assertEquals("BBBB", block.getParameter(REFERENCE));
+    }
+
+    @BeforeEach
+    void setup()
```

Comment:
This could be added at the beginning of the class, since this is actually the first one called